src/features/device-types/device-types.ts

```diff
@@ -172,7 +171,12 @@ async function fetchDeviceTypesAndReschedule(): Promise<
 		// in case another api request comes before the first completes
 		if (!deviceTypesCache) {
 			deviceTypesCache = promise;
+			// We want to clear out the cache if the cached promise fails
+			promise.catch(() => {
```

Comment:
How about moving this to a proper `catch (err) {}` below so that we don't mix await & promises and in there do something like:
```
if (deviceTypesCache === promise) {
    deviceTypesCache = undefined;
}
```
and then rethrow?
Regardless of that, let's also add a `captureException` in here as well.